types/react-navigation/index.d.ts

```diff
@@ -1248,31 +1248,31 @@ export interface NavigationInjectedProps<P = NavigationParams> {
 }
 
 // If the wrapped component is a class, we can get a ref to it
-export function withNavigation<T extends React.ComponentClass<NavigationInjectedProps>>(
-  Component: T,
-): React.ComponentType<Omit<InferProps<T>, keyof NavigationInjectedProps> & { onRef?: React.Ref<InstanceType<T>> }>;
```

Comment:
If `InferProps` returning `never` causes issues in variance, it might be better to have it return a different fallback than `never` (e.g. take another type parameter that it returns instead of never):

```ts
export type InferProps<T extends React.ComponentType<any>, Fallback> = T extends React.ComponentType<infer P> ? P : Fallback;
```

Reply:
The original problem here was that when `strictFunctionTypes` is enabled and `T` is a component class with a props type that extends `NavigationInjectedProps` with additional properties, `T` won't extend `React.ComponentClass<NavigationInjectedProps>`.  I don't see how your comment relates to that problem.